frontend-ui/src/views/PipelineView.vue

```diff
@@ -167,8 +170,29 @@ async function loadData(limit: number, skip: number, filter?: string) {
 }
 
 async function loadLastRuns() {
-  // TODO: Load last runs
+  const schedule_names = Array.from(new Set(tasks.value.map((task) => task.schedule_name)))
+  loadingStore.startLoading('Loading last runs...')
+
+  // Set lastRunsLoaded to true immediately so UI can start displaying data as it arrives
   lastRunsLoaded.value = true
+
+  // Process each schedule individually with a pause between requests
+  for (const schedule_name of schedule_names) {
+    if (schedule_name) {
+      try {
+        const result = await scheduleStore.fetchSchedule(schedule_name)
+        if (result && result.most_recent_task) {
+          schedulesLastRuns.value[schedule_name] = result.most_recent_task
+        }
+        // Add a 100ms pause between requests to avoid overwhelming the backend
+        await new Promise(resolve => setTimeout(resolve, 100))
+      } catch (error) {
+        console.error(`Failed to fetch schedule ${schedule_name}:`, error)
```

Comment:
Would be nice to display an error label (without details) in place of the last run information in the UI, so that it is clearer that UI attempted to load this but failed. Could be done later again.

Reply:
Added in current PR instead  so it would be reusable